File: frontend/kitchen/src/types/order.ts
export const OrderStatus = {
  Received: 'Received',
  Preparing: 'Preparing',
  Ready: 'Ready',
  Served: 'Served',
  Closed: 'Closed'
} as const;

export type OrderStatus = typeof OrderStatus[keyof typeof OrderStatus];

export interface OrderItem {
  id: number;
  menuItemId: number;
  menuItemName: string;
  quantity: number;
  unitPrice: number;
  specialInstructions?: string;
}

export interface Order {
  id: number;
  tableId: number;
  tableNumber: number;
  status: OrderStatus;
  subTotal: number;
  taxAmount: number;
  totalAmount: number;
  specialInstructions?: string;
  items: OrderItem[];
  createdAt: string;
  updatedAt: string;
}

export interface OrderStatusUpdate {
  orderId: number;
  status: OrderStatus;
  timestamp: string;
}

export interface KitchenFilters {
  status: OrderStatus | 'All';
  searchTerm: string;
}

export interface KitchenState {
  orders: Order[];
  filters: KitchenFilters;
  isLoading: boolean;
  error: string | null;
  connectionStatus: 'connected' | 'disconnected' | 'connecting';
}

File: backend/src/TableOrder.Backend/Models/Order.cs
namespace TableOrder.Backend.Models;

public class Order
{
    public int Id { get; set; }
    public int TableId { get; set; }
    public OrderStatus Status { get; set; } = OrderStatus.Received;
    public decimal SubTotal { get; set; }
    public decimal TaxAmount { get; set; }
    public decimal TotalAmount { get; set; }
    public string? SpecialInstructions { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

    // Navigation properties
    public Table Table { get; set; } = null!;
    public ICollection<OrderItem> OrderItems { get; set; } = new List<OrderItem>();
    public ICollection<Payment> Payments { get; set; } = new List<Payment>();
}

public enum OrderStatus
{
    Received,
    Preparing,
    Ready,
    Served,
    Closed
}
